operator/api/common/common_types.go

```diff
@@ -49,6 +49,10 @@ type Image struct {
 	//+kubebuilder:validation:Optional
 	Tag string `json:"tag" default:"latest"`
 
+	// The digest (version) of the image
+	//+kubebuilder:validation:Optional
+	Digest string `json:"digest,omitempty"`
```

Comment:
A check is needed in the hooks to verify that either digest or tag should be provided